specs/_features/eip7547/beacon-chain.md

```diff
@@ -0,0 +1,273 @@
+    state_root: Bytes32
+    receipts_root: Bytes32
+    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    prev_randao: Bytes32  # 'difficulty' in the yellow paper
+    block_number: uint64  # 'number' in the yellow paper
+    gas_limit: uint64
+    gas_used: uint64
+    timestamp: uint64
+    extra_data: ByteList[MAX_EXTRA_DATA_BYTES]
+    base_fee_per_gas: uint256
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_PAYLOAD]
+    withdrawals: List[Withdrawal, MAX_WITHDRAWALS_PER_PAYLOAD]
+    blob_gas_used: uint64
+    excess_blob_gas: uint64
+    previous_inclusion_list_summary: InclusionListSummary  # [New in EIP7547]
+```
+
+#### `ExecutionPayloadHeader`
+
+Adding `previous_inclusion_list_summary_root`.
+
+```python
+class ExecutionPayloadHeader(Container):
+    # Execution block header fields
+    parent_hash: Hash32
+    fee_recipient: ExecutionAddress
+    state_root: Bytes32
+    receipts_root: Bytes32
+    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    prev_randao: Bytes32
+    block_number: uint64
+    gas_limit: uint64
+    gas_used: uint64
+    timestamp: uint64
+    extra_data: ByteList[MAX_EXTRA_DATA_BYTES]
+    base_fee_per_gas: uint256
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
+    transactions_root: Root
+    withdrawals_root: Root
+    blob_gas_used: uint64
+    excess_blob_gas: uint64 
+    previous_inclusion_list_summary_root: Root  # [New in EIP7547]
+```
+
+## Beacon chain state transition function
+
+### Execution engine
+
+#### Engine APIs
+
+##### New `verify_and_notify_new_inclusion_list`
+
+```python
+def verify_and_notify_new_inclusion_list(self: ExecutionEngine,
+                              inclusion_list: InclusionList) -> bool:
+    """
+    Return ``True`` if and only if the transactions in the inclusion list can be successfully executed
+    starting from the execution state corresponding to the `parent_hash` in the inclusion list 
+    summary.
+    """
+    ...
+```
+
+##### Modified `verify_and_notify_new_payload`
+
+Add `is_inclusion_list_available` check.
+
+```python
+def verify_and_notify_new_payload(self: ExecutionEngine,
+                                  new_payload_request: NewPayloadRequest) -> bool:
+    """
+    Return ``True`` if and only if ``new_payload_request`` is valid with respect to ``self.execution_state``.
+    """
+    execution_payload = new_payload_request.execution_payload
+    parent_beacon_block_root = new_payload_request.parent_beacon_block_root
+
+    # [Modified in Deneb:EIP4788]
+    if not self.is_valid_block_hash(execution_payload, parent_beacon_block_root):
+        return False
+
+    # [New in Deneb:EIP4844]
+    if not self.is_valid_versioned_hashes(new_payload_request):
+        return False
+
+    # [New in EIP7547]
+    if not self.is_inclusion_list_available(new_payload_request):
+        return False
```

Comment:
I believe we can remove this check. As discussed, we will base IL filtering solely on the fork choice block tips.

Reply:
ty